### PR TITLE
revert dropUntil*() for Stream and List

### PR DESCRIPTION
**Pull request by ruslansennov**
see #2007

### CODE REVIEW

File: vavr/src/main/java/io/vavr/collection/Collections.java [hunk 1]
```diff
@@ -51,7 +51,9 @@ static boolean areEqual(Iterable<?> iterable1, Iterable<?> iterable2) {
                     .foldLeft(Iterator.of(empty), (product, ignored) -> product.flatMap(el -> seq.map(t -> (S) el.append(t))));
         }
     }
-    
+
+    // DEV-NOTE: Use this method for non-infinite and direct-access collection only
+    // see https://github.com/vavr-io/vavr/issues/2007
```
Comment (gtrefs) [inline vavr/src/main/java/io/vavr/collection/Collections.java]: Could you rather add a little explanation why one should not use this method when using infinite collections? I think this is more convinient than referring to a github issue which might get lost over the time.

File: vavr/src/main/java/io/vavr/collection/Collections.java [hunk 2]
```diff
@@ -63,6 +65,8 @@ static boolean areEqual(Iterable<?> iterable1, Iterable<?> iterable2) {
         return (S) seq.take(0);
     }
 
+    // DEV-NOTE: Use this method for non-infinite and direct-access collection only
+    // see https://github.com/vavr-io/vavr/issues/2007
```
Comment (gtrefs) [inline vavr/src/main/java/io/vavr/collection/Collections.java]: See comment above.

File: vavr/src/main/java/io/vavr/collection/Collections.java (hunk 1 above)
Comment (ruslansennov) [inline vavr/src/main/java/io/vavr/collection/Collections.java]: knock on wood :smile: 
I added a short description

**Comment by ruslansennov** [comment]
> [Queue|List|Stream].drop[While|Until]
[Queue|List|Stream].dropRightUntil

these tests already exist, see `AbstractTraversableTest`

**Pull request merged**